### ISSUE TITLE
calculation issue in hexagon API

### ISSUE THREAD

**Issue by ChaosRez** [basonjui/hexagon-grid-system#20]
Hello, 
thanks for your work.
I have this sample which result a non-symmetric hexagon

```JSON
{
    "latitude": 52.513337,
    "longitude": 13.326161,
    "radius": 10000
}
```

**Comment by basonjui** [comment]
Hello @ChaosRez ,

Apology for the very late reply... I've been caught up with work and do not have much free time to check on my GitHub. Thank you for raising this issue, I always anticipated this but forgot to add it to the README. 

However, this is **not a bug** of hexagon-grid-system. To explain it simply, it has a lot to do with **map projection** and not much to do with hexagon calculations.

### OpenStreetMap (OSM) and Spherical Pseudo-Mercator projection

I was able to reproduce your issue on https://geojson.io, and if you look to the bottom of the screenshot where I highlighted in red, you can see that _geojson.io_ is using OpenStreetMap as their map layer.

![Screenshot 2023-12-31 at 1 29 33 AM](https://github.com/basonjui/hexagon-grid-system/assets/60636087/7ef06586-0d6e-45ed-927d-268f63a68a84)

Now, OpenStreetMap uses a very common map projection called [Spherical Pseudo-Mercator projection](https://wiki.openstreetmap.org/wiki/Mercator#Spherical_Pseudo-Mercator_projection#:~:text=uses%20a%20Pseudo-Mercator%20projection) -  where the Earth is modelized **as if it was a perfect sphere** (which it is not).


### Problem with Spherical Pseudo-Mercator projection

TL;DR
Any polygon that is being displayed on a map using Pseudo-Mercator projection **will be grossly distorted** - the further it is away from the Equator (or closer it is to the North and South poles).

Well explained by [Wikipedia](https://en.wikipedia.org/wiki/Web_Mercator_projection#Advantages_and_disadvantages#:~:text=polar%20regions%20are%20grossly%20exaggerated), the Spherical Mercator projection has the following **properties**: 
- North is up everywhere
- Meridians are equally spaced vertical lines
- Angles are locally correct (assuming spherical coordinates)
- And areas inflate with distance from the equator such that the **polar regions are grossly exaggerated** (this is the problem).


### What does this mean?

The hexagon calculations are good (not perfect, but it is decently good). I calculate the hexagons using OpenStreetMap's [Spherical Pseudo-Mercator projection algorithm](https://wiki.openstreetmap.org/wiki/Mercator#Spherical_Pseudo-Mercator_projection).

It can be off by about 0.5%, but that error-margin seems acceptable by most map services/platforms (including OpenStreetMap).
